frontend/src/views/Forms/FormView.jsx

```diff
@@ -0,0 +1,173 @@
+// TODO: Example view! Replace with actual business logic.
+
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import {
+  Box,
+  Typography,
+  Paper,
+  CircularProgress,
+  Alert,
+  Container
+} from '@mui/material'
+import axios from 'axios'
+import { useKeycloak } from '@react-keycloak/web'
+import { CONFIG } from '@/constants/config'
+
+const FormView = () => {
+  const { formSlug, linkKey } = useParams()
+  const { keycloak } = useKeycloak()
+  const [formData, setFormData] = useState(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+
+  // Determine if this is an authenticated or anonymous access
+  const isAnonymousAccess = Boolean(linkKey)
+
+  useEffect(() => {
+    const fetchFormData = async () => {
+      try {
+        setLoading(true)
+        setError(null)
+
+        console.log('Fetching form data for:', {
+          formSlug,
+          linkKey,
+          isAnonymousAccess,
+          keycloakAuthenticated: keycloak.authenticated,
+          hasToken: !!keycloak.token
+        })
+
+        // Use different API endpoints based on access type
+        const apiUrl = isAnonymousAccess
+          ? `${CONFIG.API_BASE}/forms/${formSlug}/${linkKey}`
+          : `${CONFIG.API_BASE}/forms/${formSlug}`
+
+        const axiosConfig = {
+          timeout: 10000 // 10 second timeout
+        }
+
+        // For authenticated access, we need to include auth headers
+        if (!isAnonymousAccess && keycloak.authenticated) {
+          axiosConfig.headers = {
+            Authorization: `Bearer ${keycloak.token}`
+          }
+          console.log(
+            'Adding auth header with token:',
+            keycloak.token ? 'present' : 'missing'
+          )
+        }
+
+        console.log('API URL:', apiUrl)
+        console.log('Axios config:', axiosConfig)
+
+        const response = await axios.get(apiUrl, axiosConfig)
+
+        console.log('Form data response:', response)
+        setFormData(response.data)
+      } catch (err) {
+        console.error('Error fetching form data:', err)
+        console.error('Error response:', err.response)
+        console.error('Error status:', err.response?.status)
+        console.error('Error data:', err.response?.data)
+
+        let errorMessage = 'Failed to load form'
+        if (err.response?.data?.detail) {
```

Comment:
the if else logic gets a little complicated here. Please clean it up with a status to message mapping method.